File: wup1/Python/SleepIn.py
'''
    sleep_in
    The parameter weekday is True if it is a weekday, 
    and the parameter vacation is True if we are on vacation. 
    We sleep in if it is not a weekday or we're on vacation. 
    Return True if we sleep in.
'''

def sleep_in(weekday, vacation):
    if not weekday or vacation:
        return True
    else:
        return False
        
def tests():
    assert (sleep_in(False,False) == True)
    assert (sleep_in(True, False) == False)
    assert (sleep_in(False, True) == True)
    assert (sleep_in(True, True) == True)
    print("All tests passed")
    
tests()

File: wup1/Java/SleepIn.java
/*  SleepIn
    The parameter weekday is true if it is a weekday, 
    and the parameter vacation is true if we are on vacation. 
    We sleep in if it is not a weekday or we're on vacation. 
    Return true if we sleep in.*/
public class SleepIn {
    
    public static void main(String[] args) {
        SleepIn si = new SleepIn();
        si.test();
	}

    public boolean sleepIn(boolean weekday, boolean vacation) {
        if (!weekday || vacation) 
            return true;
        else    
            return false;
    }

    public void test() {
        assert sleepIn(false,false) == true;
        assert sleepIn(true, false) == false;
        assert sleepIn(false, true) == true;
        assert sleepIn(true, true) == true;
        System.out.println("All tests passed");
    }
}